src/components/Match/MatchContainer.js

```diff
@@ -148,12 +149,37 @@ const MatchContainer = ( ) => {
     onlineSuggestionsAttempted
   } );
 
-  const onTaxonChosen = useCallback( selectedTaxon => {
-    dispatch( {
-      type: "SELECT_TAXON",
-      selectedTaxon
-    } );
-  }, [] );
+  const onSuggestionChosen = useCallback( selection => {
+    const suggestionsList = [...orderedSuggestions];
+
+    // make sure to reorder the list by confidence score
+    // for when a user taps multiple suggestions and pushes a new top suggestion to
+    // the top of the list
+    const sortedList = _.orderBy(
+      suggestionsList,
+      suggestion => suggestion.combined_score || suggestion.score,
+      ["desc"]
+    );
+
+    // order the chosen suggestion at the beginning of the list, so it
+    // can become the new top suggestion
+    const chosenIndex = _.findIndex(
+      sortedList,
+      s => s === selection
```

Comment:
This use of object equivalence strikes me as a little risky. For example, if you wanted to use truly immutable data here and replaced

```javascript
const suggestionsList = [...orderedSuggestions];
```

with

```javascript
const suggestionsList = _.cloneDeep( orderedSuggestions );
```

it would not work. It might be a little more defensive to do something like `suggestion => suggestion.taxon.id === selection.taxon.id`.